pennylane/resource/second_quantization.py

```diff
@@ -122,10 +122,12 @@ def __init__(
 
         self.n = two_electron.shape[0] * 2
 
-        self.factors, self.eigvals, self.eigvecs = factorize(
+        self.factors, _, self.eigvecs = factorize(
             self.two_electron, self.tol_factor, self.tol_eigval
         )
 
+        feigvals = np.linalg.eigvalsh(self.factors)
+        self.eigvals = [eigvals[np.where(np.abs(eigvals) > tol_eigval)] for eigvals in feigvals]
```

Comment:
Why this is changed?

Reply:
Just using numpy here to use vectorization. Logic is the same.